### DISCUSSION TITLE
When applying component tokens, they don't apply the corresponding colour style

### DISCUSSION THREAD

**Discussion by ananth22by7** [six7/figma-tokens#346] [Q&A]
Say I have a global token called `color-black` whose value is `#000000`.
I also have a corresponding color style called `color-black` with the same value.
When I apply the token via the plugin, Figma correctly applies the corresponding color style too.

Now if I have a component token called `dark-surface` that points to token `color-black`, and I apply that token to a layer fill, Figma unfortunately applies only the raw value `#000000` to the layer and it doesn't recognise the corresponding color-style.

Is this expected behaviour?

**Comment by TheIcarusWings** [comment]
You need to create a color style with the exact same name. This way Figma will apply the(named) color style when you apply the token. Afterward, any time you update the token, it will update the color style. Make sure you have "update styles" selected.

**Reply by ananth22by7** [reply]
Thanks for the quick reply. If I am following you correctly, this is what I've done, but the problem is not with this.

This is working for global tokens that directly point to the primitive value. However, for alias tokens that point to global tokens which are pointing to the primitive value, this doesn't work. Here is a screen recording to illustrate:

https://user-images.githubusercontent.com/9886093/144034171-91e73d03-6fd8-4ee1-89ce-b5a0591f673c.mov

**Reply by TheIcarusWings** [reply]
You need to create the style for that token too.

**Reply by TheIcarusWings** [reply]
Make sure you follow the same folder structure on both sides. `color-bg.color-bg-live` on token and `color-bg/color-bg-live` on figma.
